lib/ansible/module_utils/aws.py

```diff
@@ -0,0 +1,123 @@
+**aggressively as the exact correct interface for ansible AWS modules**
+**is identified.  In particular, until this notice goes away or is**
+**changed, methods may disappear from the interface.  Please don't**
+**publish modules using this except directly to the main Ansible**
+**development repository.**
+
+In order to use this module, include it as part of a custom
+module as shown below.
+
+  from ansible.module_utils.cloud import AnsibleAWSModule
+  m=AnsibleAWSModule(argument_spec=dictionary, supports_check_mode=boolean
+                     mutually_exclusive=list1, required_together=list2)
+
+The 'AnsibleAWSModule' module provides similar, but more restricted,
+interfaces to the normal Ansible module.  It also includes the following additional methods.
+
+  m.fail_json_aws(Exception) # - take an exception and make a decent failure
+  m.get_conn() # - get an AWS connection. 
+
+"""
+
+from ansible.module_utils.basic import AnsibleModule
+from ansible.module_utils.ec2 import HAS_BOTO3, camel_dict_to_snake_dict
+import traceback
+
+#We will also export HAS_BOTO3 so end user modules can use it. 
+HAS_BOTO3 = HAS_BOTO3
+
+class AnsibleAWSModule(object):
+    """An ansible module class for AWS modules
+    
+    AnsibleAWSModule provides an a class for building modules which
+    connect to Amazon Web Services.  The interface is currently more
+    restricted than the basic module class with the aim that later the
+    basic module class can be reduced.  If you find that any key
+    festure is missing please contact the author/Ansible AWS team
+    (available on #ansible-aws on IRC) to request the additional
+    features needed.
+    """
+    
+    def __init__(self, **kwargs):
+        try:
+            default_args=kwargs["default_args"]
+            del kwargs["default_args"]
+        except KeyError:
+            default_args=True
+
+        try:
+            autoretry=kwargs["autoretry"]
+            del kwargs["autoretry"]
+        except KeyError:
+            autoretry=True
+
+        if default_args:
+            pass #We will set default arguments for AWS module here?
+
+        junk=None
+        jook=None
+        self._module = AnsibleModule(**kwargs)
+        self.check_mode = self._module.check_mode
+
+    @property
+    def params(self):
+        return self._module.params
+
+    def exit_json(self, *args, **kwargs):
+        return self._module.exit_json(*args, **kwargs)
+
+    def fail_json(self, *args, **kwargs):
+        return self._module.fail_json(*args, **kwargs)
+
+    def fail_json_aws(self, exception, msg=None):
+        """call fail_json with processed exception
+
+        function for converting exceptions thrown by AWS SDK modules,
+        botocore, boto3 and boto, into nice error messages.
+        """
+        last_traceback = traceback.format_exc()
+
+        try:
+            except_msg = exception.message
+        except AttributeError:
+            except_msg = str(exception)
```

Comment:
Using to_native is slightly better here (prevents tracebacks on python2 in case the exception contains non-ascii characters.)